src/test/java/seedu/address/storage/JsonSerializableTutorsPetTest.java

```diff
@@ -55,10 +93,46 @@ public void toModelType_invalidClassFile_throwsIllegalValueException() throws Ex
         assertThrows(IllegalValueException.class, dataFromFile::toModelType);
     }
 
+    /**
+     * Ensures that Tutor's Pet will not be able to boot up although Java tries
+     * to pad missing UUID digits in broken UUIDs with zeros.
+     */
+    @Test
+    public void toModelType_invalidStudentUuidInClassFile1_throwsIllegalValueException() throws Exception {
+        JsonSerializableTutorsPet dataFromFile = JsonUtil.readJsonFile(INVALID_STUDENT_UUID_IN_CLASS_FILE_1,
+                JsonSerializableTutorsPet.class).get();
+        assertThrows(IllegalValueException.class, dataFromFile::toModelType);
+    }
+
+    /**
+     * Ensures that Tutor's Pet will not be able to boot up given an incomplete UUID.
+     * (e.g "0c527a3f-8a6f-4c16-b57d-")
+     */
+    @Test
+    public void toModelType_invalidStudentUuidInClassFile2_throwsIllegalValueException() throws Exception {
+        JsonSerializableTutorsPet dataFromFile = JsonUtil.readJsonFile(INVALID_STUDENT_UUID_IN_CLASS_FILE_2,
+                JsonSerializableTutorsPet.class).get();
+        assertThrows(IllegalValueException.class, dataFromFile::toModelType);
+    }
+
     @Test
     public void toModelType_duplicateClasses_throwsIllegalValueException() throws Exception {
         JsonSerializableTutorsPet dataFromFile = JsonUtil.readJsonFile(DUPLICATE_CLASS_FILE,
                 JsonSerializableTutorsPet.class).get();
         assertThrows(IllegalValueException.class, dataFromFile::toModelType);
     }
+
+    /**
+     * If Tutor's Pet encounters duplicate student UUIDs in a class, it will not load
+     * duplicate UUIDs into the model. Hence, Tutor's Pet should still boot up successfully.
```

Comment:
```suggestion
     * If Tutor's Pet encounters duplicate {@code Student UUID}s in a class, it will not load
     * duplicate {@code UUID}s into the model. Hence, Tutor's Pet should still boot up successfully.
```